src/software/backend/simulator_backend.cpp

```diff
@@ -13,13 +14,27 @@ SimulatorBackend::SimulatorBackend(
     SimulatorBackend::SimulationSpeed simulation_speed_mode)
     : physics_time_step(physics_time_step),
       world_time_increment(world_time_increment),
-      simulation_speed_mode(simulation_speed_mode)
+      simulation_speed_mode(simulation_speed_mode),
+      primitive_buffer(primitive_buffer_size),
+      in_destructor(false),
+      simulation_thread_started(false)
 {
 }
 
+SimulatorBackend::SimulatorBackend()
+    : SimulatorBackend(Duration::fromMilliseconds(5), Duration::fromSeconds(1.0 / 30.0),
```

Comment:
I'm somewhat inclined to leave it so that anyone modifying the class will use the right member variables set by the constructor. Otherwise people might use that constant directly which would then not work if a different parameter was provided. Thoughts?